packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -245,20 +304,35 @@ class UpgradeCommandRunner {
       final String errorString = e.toString();
       if (errorString.contains('fatal: HEAD does not point to a branch')) {
         throwToolExit(
-          'You are not currently on a release branch. Use git to '
-          'check out an official branch (\'stable\', \'beta\', \'dev\', or \'master\') '
-          'and retry, for example:\n'
-          '  git checkout stable'
+          'Unable to upgrade Flutter: HEAD does not point to a branch(Are you '
+          'in a detached HEAD state?).\n'
+          'Use "flutter channel" to switch to an official channel '
+          '("stable", "beta", "dev", or "master") and retry, for example:\n\n'
+          '  flutter channel stable\n\n'
+          'As an alternative, if you are okay with losing local changes you made '
+          'to the SDK, re-install Flutter by going to https://flutter.dev/docs/get-started/install.'
         );
       } else if (errorString.contains('fatal: no upstream configured for branch')) {
+        // Get the name of local branch to show in error message
+        String localBranch;
+        try {
+          localBranch = localVersion.getBranchName();
+        } on Exception catch (e) {
+          throwToolExit(e.toString());
+        }
         throwToolExit(
-          'Unable to upgrade Flutter: no origin repository configured. '
-          'Run \'git remote add origin '
-          'https://github.com/flutter/flutter\' in $workingDirectory');
+          'Unable to upgrade Flutter: No upstream repository configured for branch.\n'
+          'Run "git remote add origin https://github.com/flutter/flutter.git" and '
```

Comment:
I think we should omit these git commands, and just direct users toward the website. If they know git, telling them they don't have an upstream configured should be enough to get them on the right track, and suggesting git commands to users who aren't familiar with it could go badly.

Reply:
Omitting git commands might leave ambiguity about the problem. That will lead to the exact use case of #49713. At least showing them the git commands will make them realize what should be done.

For users uncomfortable in working with git, they will skip the git instructions and re-install anyway from the download link provided.

Reply:
We can tell them they don't have an upstream configured for the current branch without suggesting git commands for them to use.

Reply:
Should probably mention `"git branch"` alongwith `"git remote"` too while we are at it.